lang/java/ipc/src/test/java/org/apache/avro/specific/TestSpecificDatumWriter.java

```diff
@@ -50,4 +52,26 @@ public void testResolveUnion() throws IOException {
     assertEquals(expectedJson, out.toString("UTF-8"));
   }
 
+  @Test
+  public void testIncompleteRecord() throws IOException {
+    final SpecificDatumWriter<TestRecord> writer = new SpecificDatumWriter<>();
+    Schema schema = TestRecord.SCHEMA$;
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    JsonEncoder encoder = EncoderFactory.get().jsonEncoder(schema, out);
+
+    writer.setSchema(schema);
+
+    TestRecord testRecord = new TestRecord();
+    testRecord.setKind(Kind.BAR);
+    testRecord.setHash(new MD5(new byte[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5 }));
+
+    try {
+      writer.write(testRecord, encoder);
+      fail("Exception not thrown");
+    } catch (NullPointerException e) {
+      assertTrue(e.getMessage().contains("null of string in field name"));
```

Comment:
`... in field name` sounds like something is wrong in the field's name.
IMO it would be clearer if the field name is wrapped in single quotes: `... in field 'name'`

Reply:
Thanks a lot for taking a look and your suggestion. It would be clearer indeed. I've made a change for this.